trieste/models/gpflow/models.py

```diff
@@ -57,8 +57,28 @@ def __init__(
             )
         self._num_kernel_samples = num_kernel_samples
 
+        self._ensure_variable_model_data()
+
+    def __repr__(self) -> str:
+        """"""
+        return f"GaussianProcessRegression({self._model!r}, {self.optimizer!r})"
+
+    @property
+    def model(self) -> GPR | SGPR:
+        return self._model
+
+    def _ensure_variable_model_data(self) -> None:
         # GPflow stores the data in Tensors. However, since we want to be able to update the data
         # without having to retrace the acquisition functions, put it in Variables instead.
+        # To tell if the data is stored in a way it can be updated without retracing,
```

Comment:
I don't think this comment quite explains why we need this. Maybe something like:

> Following seriasliation-deserialisation, the dynamic Variable shape can get overwritten (preventing  subsequent changes to the data) so we also ensure it's correct before every update.

Reply:
It's not only about serialization though. I've merged the two description into a single comment